src/libvirtApi/common.ts

```diff
@@ -503,29 +414,102 @@ export const initState = (): Promise<[void, void]> => Promise.all([
+        const [info] = await call<[DBusProps]>(vm.connectionName, vm.id, "org.libvirt.Domain", "GetStats", [flags, 0], { timeout: 5000, type: "uu" });
+        if (Object.getOwnPropertyNames(info).length > 0) {
+            const props: Partial<VM> = { name: vm.name, connectionName: vm.connectionName, id: vm.id };
+            let avgvCpuTime = 0;
+
+            if ("balloon.rss" in info)
+                props.rssMemory = get_number_prop(info, "balloon.rss");
+            else if ("state.state" in info && get_number_prop(info, "state.state") == Enum.VIR_DOMAIN_SHUTOFF)
+                props.rssMemory = 0.0;
+            for (let i = 0; i < get_number_prop(info, "vcpu.maximum"); i++) {
+                if (!(`vcpu.${i}.time` in info))
+                    continue;
+                avgvCpuTime += get_number_prop(info, `vcpu.${i}.time`);
+            }
+            avgvCpuTime /= get_number_prop(info, "vcpu.current");
+            if (get_number_prop(info, "vcpu.current") > 0) {
+                props.actualTimeInMs = Date.now();
+                props.cpuTime = avgvCpuTime;
+            }
+            props.disksStats = calculateDiskStats(info);
+
+            logDebug(`pollVmUsage: ${JSON.stringify(props)}`);
+            store.dispatch(updateVm(props));
+        }
+    } catch (ex) {
+        console.warn(`GetStats(${vm.name}, ${vm.connectionName}) failed: ${String(ex)}`);
+    }
 }
 
-export function usageStopPolling({
-    name,
-    connectionName
-} : {
-    name: string,
-    connectionName: ConnectionName,
-}): void {
-    store.dispatch(updateVm({
-        connectionName,
-        name,
-        usagePolling: false
-    }));
+export async function pollUsageNow() {
+    const vms = store.getState().vms;
+
+    if (cockpit.hidden)
+        return;
+
+    if (typeof usagePolling === "string") {
+        const vm = vms.find(vm => vm.uuid === usagePolling);
+        if (vm)
+            await pollVmUsage(vm);
+    } else if (usagePolling === true) {
+        for (const vm of vms) {
+            if (usagePolling !== true || cockpit.hidden)
+                break;
+            await pollVmUsage(vm);
+        }
+    }
+}
+
+let usagePollingTimeoutId: number = 0;
+
+async function startUsagePolling() {
+    // We don't use setInterval since pollUsageNow might take
+    // considerable time itself. So we only restart the timer once
+    // polling is done.
+    await pollUsageNow();
+    if (usagePolling !== false && !cockpit.hidden && usagePollingTimeoutId == 0) {
+        usagePollingTimeoutId = window.setTimeout(() => {
+            usagePollingTimeoutId = 0;
+            startUsagePolling();
+        }, store.getState().config.refreshInterval);
+    }
+}
+
+function cancelUsagePolling() {
+    if (usagePollingTimeoutId) {
+        window.clearTimeout(usagePollingTimeoutId);
+        usagePollingTimeoutId = 0;
+    }
+}
+
+cockpit.addEventListener("visibilitychange", async () => {
+    if (cockpit.hidden)
+        cancelUsagePolling();
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2183-d36c4c17-20250905-135613-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)